### PR TITLE
Pass trusted peers to config

### LINKED ISSUE #364: node: Light node can be initialised with multiple trusted peers
### Summary

Light node needs more options to include in trusted peers instead of 1

### Impact on users

In current implementation, a light node can be only initialised with 1 trusted peer. It will be beneficial to allow multiple trusted peers (array of them) for light node initialisation

### PR DESCRIPTION
**Pull request by vgonkivs**
Fixes #364 

Chages:
* multiple trusted peers can be passed on node initialisation;
* trusted peer will be taken randomly on `performRequest`;

### CODE REVIEW

File: cmd/flags_header.go [hunk 1]
```diff
@@ -12,8 +12,8 @@ import (
 )
 
 var (
-	headersTrustedHashFlag = "headers.trusted-hash"
-	headersTrustedPeerFlag = "headers.trusted-peer"
+	headersTrustedHashFlag  = "headers.trusted-hash"
+	headersTrustedPeersFlag = "headers.trusted-peers"
```
Comment (Wondertan) [inline cmd/flags_header.go:16]: This is a breaking change so we should add the proper entry to changelog

File: node/services/config.go [hunk 2]
```diff
@@ -17,32 +17,40 @@ type Config struct {
 	// TrustedHash is the Block/Header hash that Nodes use as starting point for header synchronization.
 	// Only affects the node once on initial sync.
 	TrustedHash string
-	// TrustedPeer is the peer we trust to fetch headers from.
+	// TrustedPeers is the peer we trust to fetch headers from.
```
Comment (Wondertan) [inline node/services/config.go:20]: Should also change the comment accordingly to plural form

File: service/header/p2p_exchange.go [hunk 3]
```diff
@@ -24,13 +25,13 @@ type P2PExchange struct {
 
 	// TODO @renaynay: post-Devnet, we need to remove reliance of Exchange on one bootstrap peer
 	// Ref https://github.com/celestiaorg/celestia-node/issues/172#issuecomment-964306823.
-	trustedPeer peer.ID
+	trustedPeers peer.IDSlice
```
Comment (Wondertan) [inline service/header/p2p_exchange.go:28]: You can remove the TODO above.

File: service/header/p2p_exchange.go [hunk 4]
```diff
@@ -98,11 +99,10 @@ func (ex *P2PExchange) performRequest(ctx context.Context, req *pb.ExtendedHeade
 	if req.Amount == 0 {
 		return make([]*ExtendedHeader, 0), nil
 	}
-	if ex.trustedPeer == "" {
-		return nil, fmt.Errorf("no trusted peer")
-	}
```
Comment (Wondertan) [inline service/header/p2p_exchange.go:103]: `if len(ex.trustedPeer) == 0` ?

File: service/header/p2p_exchange_test.go [hunk 5]
```diff
@@ -109,17 +110,17 @@ func createMocknet(ctx context.Context, t *testing.T) (libhost.Host, libhost.Hos
...
 	err := serverSideEx.Start(context.Background())
 	require.NoError(t, err)
 
 	t.Cleanup(func() {
 		serverSideEx.Stop(context.Background()) //nolint:errcheck
 	})
-
-	return NewP2PExchange(host, peer.ID()), store
+	ids := []peer.ID{tpeer.ID()}
+	return NewP2PExchange(host, ids), store
```
Comment (Wondertan) [inline service/header/p2p_exchange_test.go:123]: Inline those?

**Review by Wondertan — COMMENTED** [review]
Thanks

File: service/header/p2p_exchange.go (hunk 4 above)
Comment (vgonkivs) [reply service/header/p2p_exchange.go:103]: I thought now we are using peers by default if nothing were provided...

**Review by liamsi — COMMENTED** [review]
Hlib's comments are on point. I do not have anything to add :)

File: cmd/flags_header.go (hunk 1 above)
Comment (vgonkivs) [reply cmd/flags_header.go:16]: done

File: service/header/p2p_exchange.go (hunk 3 above)
Comment (vgonkivs) [reply service/header/p2p_exchange.go]: removed

**Review by liamsi — APPROVED** [review]
👍👏🙌👌

File: node/services/config.go [hunk 6]
```diff
@@ -17,32 +17,39 @@ type Config struct {
...
 	}
 }
 
-func (cfg *Config) trustedPeer() (*peer.AddrInfo, error) {
-	if cfg.TrustedPeer == "" {
-		log.Warn("No Trusted Peer provided. Headers won't be synced accordingly")
-		return &peer.AddrInfo{}, nil
+func (cfg *Config) trustedPeers() ([]*peer.AddrInfo, error) {
+	if len(cfg.TrustedPeers) == 0 {
+		cfg.TrustedPeers = params.Bootstrappers()
```
Comment (renaynay) [inline node/services/config.go:36]: We should log this IMO.

```
No trusted peers given, initializing with default bootstrappers as trusted peers
```

File: service/header/p2p_exchange.go [hunk 7]
```diff
@@ -98,11 +97,14 @@ func (ex *P2PExchange) performRequest(ctx context.Context, req *pb.ExtendedHeade
 	if req.Amount == 0 {
 		return make([]*ExtendedHeader, 0), nil
 	}
-	if ex.trustedPeer == "" {
+
+	if len(ex.trustedPeers) == 0 {
 		return nil, fmt.Errorf("no trusted peer")
```
Comment (renaynay) [inline service/header/p2p_exchange.go:102]: "no trusted peers"

**Review by renaynay — COMMENTED** [review]
Awesome! Some small comments.

File: service/header/p2p_exchange.go (hunk 7 above)
Comment (vgonkivs) [reply service/header/p2p_exchange.go]: updated

File: node/services/config.go (hunk 6 above)
Comment (vgonkivs) [reply node/services/config.go:37]: added log

File: service/header/p2p_exchange.go (hunk 4 above)
Comment (Wondertan) [reply service/header/p2p_exchange.go]: Yes we are, but it is still possible to import P2PExchange from another project and give zero peers.
Think of packages inside ‘service’(will probably get extracted soon) as libraries that eventually will be used by other people and not just by us.

File: node/services/config.go (hunk 6 above)
Comment (Wondertan) [reply node/services/config.go:37]: @renaynay, explain why. _Not_ passing trusted peers is going to be a default case after this PR, and there is no point to warn user that their are doing something wrong by using defaults

Comment (Wondertan) [reply node/services/config.go:37]: Though Info log is fine, unless it's not a Warning, like it was before

**Review by liamsi — APPROVED** [review]

File: node/services/config.go (hunk 6 above)
Comment (renaynay) [reply node/services/config.go:37]: No, that's what i meant. It should be an INFO log.

Comment (renaynay) [reply node/services/config.go:37]: thank you for adding the log @vgonkivs , this comment thread is now resolved.

File: service/header/p2p_exchange.go (hunk 4 above)
Comment (vgonkivs) [reply service/header/p2p_exchange.go]: condition added

**Review by Wondertan — APPROVED** [review]

**Pull request merged**